### DISCUSSION TITLE
OnInitialActivate Invalid

### DISCUSSION THREAD

**Discussion by wf-soft** [canton7/Stylet#220] [Q&A]
In the following case, screen is not the main window, but through the injected conductor, oninitialactivate will not be triggered

`
public class MainViewModel : Screen
    {
        [Inject]
        public Test2ViewModel Test2VM { get; set; }
    }

public class Test2ViewModel : Conductor<IScreen>.Collection.OneActive
    {
        protected override void OnViewLoaded()
        {
            base.OnViewLoaded();
            Items.Add(Helper.Ioc.Get<TestViewModel>());
            Items.Add(Helper.Ioc.Get<TestViewModel>());
            Items.Add(Helper.Ioc.Get<TestViewModel>());
            Items[0].DisplayName = "0";
            Items[1].DisplayName = "1";
            Items[2].DisplayName = "2";
        }
    }

public class TestViewModel : Conductor<string>.Collection.OneActive
    {
        protected override void OnInitialActivate()
        {
            base.OnInitialActivate();
            Debug.WriteLine("我被激活了");
        }
    }
`

**Comment by wf-soft** [comment]
@canton7 Is this normal?

**Comment by canton7** [comment]
I converted this to a discussion, as it is a support request.

**Comment by canton7** [comment]
TestViewModel is not used anywhere in your code.

**Comment by canton7** [comment]
You're probably after `Test2VM.ConductWith(this)`, which will activate `Test2VM` when `MainViewModel` is activated, etc. With constructor injection you'd call this in the constructor, but life is a bit more complex with property injection:

```cs
public class MainViewModel : Screen, IInjectionAware
{
    [Inject]
    public Test2ViewModel Test2VM { get; set; }

    public void PropertiesInjected()
    {
        Test2VM.ConductWith(this);
    }
}
```
